From 2fa52d61b1abdf4a3f3b153c771fb2081666430c Mon Sep 17 00:00:00 2001
From: Mingli Yu <Mingli.Yu@windriver.com>
Date: Tue, 21 May 2019 15:20:34 +0800
Subject: [PATCH] skip the test when gcc not deployed

Skip the tests which depend on gcc when
gcc not deployed.

Upstream-Status: Submitted[https://sourceware.org/ml/elfutils-devel/2019-q2/msg00091.html]

Signed-off-by: Mingli Yu <Mingli.Yu@windriver.com>

---
 tests/run-disasm-x86-64.sh | 2 ++
 tests/run-disasm-x86.sh    | 2 ++
 tests/run-strip-g.sh       | 2 ++
 tests/run-strip-nothing.sh | 2 ++
 4 files changed, 8 insertions(+)

diff --git a/tests/run-disasm-x86-64.sh b/tests/run-disasm-x86-64.sh
index 07b612b..7a32996 100755
--- a/tests/run-disasm-x86-64.sh
+++ b/tests/run-disasm-x86-64.sh
@@ -22,6 +22,8 @@ case "`uname -m`" in
   x86_64)
     tempfiles testfile45.o
     testfiles testfile45.S testfile45.expect
+    # skip the case if no gcc deployed
+    which gcc || exit 77
     ${CC} -m64 -c -o testfile45.o testfile45.S
     testrun_compare ${abs_top_builddir}/src/objdump -d testfile45.o < testfile45.expect
     ;;
diff --git a/tests/run-disasm-x86.sh b/tests/run-disasm-x86.sh
index 7ac73ad..f0d4796 100755
--- a/tests/run-disasm-x86.sh
+++ b/tests/run-disasm-x86.sh
@@ -22,6 +22,8 @@ case "`uname -m`" in
   x86_64 | i?86 )
     tempfiles testfile44.o
     testfiles testfile44.S testfile44.expect
+    # skip the case if no gcc deployed
+    which gcc || exit 77
     ${CC} -m32 -c -o testfile44.o testfile44.S
     testrun_compare ${abs_top_builddir}/src/objdump -d testfile44.o < testfile44.expect
     ;;
diff --git a/tests/run-strip-g.sh b/tests/run-strip-g.sh
index 1592121..9b8157e 100755
--- a/tests/run-strip-g.sh
+++ b/tests/run-strip-g.sh
@@ -24,6 +24,8 @@
 
 tempfiles a.out strip.out debug.out readelf.out
 
+# skip the test if gcc deployed
+which gcc || exit 77
 echo Create debug a.out.
 echo "int main() { return 1; }" | ${CC} -g -xc -
 
diff --git a/tests/run-strip-nothing.sh b/tests/run-strip-nothing.sh
index 710c200..3a81d8e 100755
--- a/tests/run-strip-nothing.sh
+++ b/tests/run-strip-nothing.sh
@@ -22,6 +22,8 @@
 
 tempfiles a.out strip.out debug.out
 
+# skip the case if no gcc deployed
+which gcc || exit 77
 # Create no-debug a.out.
 echo "int main() { return 1; }" | ${CC} -s -xc -
 
